File: packages/sdk-elixir/lib/notifique/messages.ex
defmodule Notifique.Messages do
  @moduledoc """
  Messages (templates) — POST /v1/templates/send
  Envio por template em múltiplos canais (whatsapp, sms, email).
  """

  def send(client, params, opts \\ []) do
    params = normalize_params(params)
    Notifique.request(client, :post, "/templates/send", params, opts)
  end

  defp normalize_params(params) when is_map(params) do
    params
    |> maybe_put_instance_id()
    |> maybe_put_from_name()
  end

  defp maybe_put_instance_id(params) do
    case Map.get(params, "instance_id") || Map.get(params, :instance_id) do
      nil -> params
      id ->
        params
        |> Map.delete("instance_id")
        |> Map.delete(:instance_id)
        |> Map.put("instanceId", id)
    end
  end

  defp maybe_put_from_name(params) do
    from_name = Map.get(params, "from_name") || Map.get(params, :from_name)
    from_name_api = Map.get(params, "fromName") || Map.get(params, :fromName)
    cond do
      is_binary(from_name) and is_nil(from_name_api) ->
        params
        |> Map.delete("from_name")
        |> Map.delete(:from_name)
        |> Map.put("fromName", from_name)
      true ->
        params
    end
  end
end

File: packages/sdk-php/src/Messages.php
<?php

namespace Notifique;

/**
 * Messages (templates) — POST /v1/templates/send
 */
class Messages
{
    private Notifique $client;

    public function __construct(Notifique $client)
    {
        $this->client = $client;
    }

    /**
     * POST /v1/templates/send — envio por template (whatsapp, sms, email).
     * Parâmetros (mesmos da API): to, template, variables?, channels, instanceId?, from?, fromName?
     *
     * @param array{to: list<string>, template: string, variables?: array<string, string|int>, channels: list<string>, instanceId?: string, from?: string, fromName?: string} $params
     */
    public function send(array $params): array
    {
        if (isset($params['from_name']) && !\array_key_exists('fromName', $params)) {
            $params['fromName'] = $params['from_name'];
            unset($params['from_name']);
        }
        if (isset($params['instance_id']) && !\array_key_exists('instanceId', $params)) {
            $params['instanceId'] = $params['instance_id'];
            unset($params['instance_id']);
        }
        return $this->client->request('POST', '/templates/send', $params);
    }
}
